File: src/main/java/com/example/dq/model/NullRule.java
package com.example.dq.model;

import java.util.List;

/**
 * 自定义空值规则:某列(或 * 表示所有列)的额外"视为空"取值。
 */
public record NullRule(String column, List<String> values) {

    public boolean matches(String columnName) {
        return "*".equals(column) || column != null && column.equalsIgnoreCase(columnName);
    }
}

File: dq-desktop/src/main/kotlin/com/example/dq/model/NullRule.kt
package com.example.dq.model

/**
 * 自定义空值规则:某列(或 * 表示所有列)的额外"视为空"取值。
 */
data class NullRule(val column: String?, val values: List<String>) {

    fun matches(columnName: String): Boolean {
        return "*" == column || column != null && column.equals(columnName, ignoreCase = true)
    }
}
